### PR TITLE
chakrashim: add implement v8::FunctionTemplate::Inherit

### PR DESCRIPTION
**Pull request by mybios**
I was implement v8::FunctionTemplate::Inherit and use in my game engine.

### CODE REVIEW

File: deps/chakrashim/src/v8functiontemplate.cc [hunk 1]
```diff
@@ -248,6 +249,19 @@ Local<Function> FunctionTemplate::GetFunction() {
...
     Local<Object> prototype =
       functionTemplateData->prototypeTemplate->NewInstance();
+
+    // inherit from parent
+    if (!functionTemplateData->parent.IsEmpty())
+    {
+        Local<Function> parent = functionTemplateData->parent->GetFunction();
+
+        JsValueRef parentPrototype;
+        if (JsGetProperty(*parent, iso->GetCachedPropertyIdRef(jsrt::CachedPropertyIdRef::prototype), &parentPrototype) != JsNoError)
```
Comment (jianchun) [inline deps/chakrashim/src/v8functiontemplate.cc]: nit: Please wrap lines at <= 80 columns (cpplint)

File: deps/chakrashim/src/v8functiontemplate.cc [hunk 2]
```diff
@@ -316,4 +330,16 @@ bool FunctionTemplate::HasInstance(Handle<Value> object) {
...
+{
+    void *externalData;
+    if (JsGetExternalData(this, &externalData) != JsNoError) {
+        return;
+    }
+
+    FunctionTemplateData *functionTemplateData =
+        reinterpret_cast<FunctionTemplateData*>(externalData);
+    functionTemplateData->parent = parent;
+
```
Comment (jianchun) [inline deps/chakrashim/src/v8functiontemplate.cc]: nit pick: please remove this empty line

**Comment by jianchun** [comment]
Change looks good to me! Thanks for filing the issue and providing the fix.

**Comment by mybios** [comment]
OK , Thank you.

**Comment by mybios** [comment]
I was create a single commit PR( https://github.com/Microsoft/node/pull/19 ). Please close this PR .

**Pull request closed without merging**